R/app_utilities.R

```diff
@@ -134,84 +131,89 @@ amounts_net_plot <-
-    
+                           hovertemplate = paste("FTC: %{y:$,.0f}<extra></extra>"))
+
     if("IWTC" %in%  display_cols)
       p <- p %>% add_trace(data = Y, x = ~gross_wage1_annual, type = 'scatter', mode = 'none',
                            y = ~IWTC_abated, name = 'IWTC',
                            fillcolor = set_colours[4], stackgroup = 'two',
-                           hovertemplate = paste("IWTC: %{y:$,.0f}<extra></extra>"))  
-    
+                           hovertemplate = paste("IWTC: %{y:$,.0f}<extra></extra>"))
+
     if("Accomodation Supplement" %in%  display_cols)
       p <- p %>%   add_trace(data = Y, x = ~gross_wage1_annual, type = 'scatter', mode = 'none',
                              y = ~AS_Amount,
                              name = 'Accomodation Supplement',
                              fillcolor = set_colours[3], stackgroup = 'two',
-                             hovertemplate = paste("Accomodation Supplement: %{y:$,.0f}<extra></extra>"))  
-    
+                             hovertemplate = paste("Accomodation Supplement: %{y:$,.0f}<extra></extra>"))
+
     if("Winter Energy"  %in%  display_cols )
       p <- p %>% add_trace(data = Y, x = ~gross_wage1_annual, type = 'scatter', mode = 'none',
                            y = ~WinterEnergy, name = 'Winter Energy',
                            fillcolor = set_colours[2], stackgroup = 'two',
-                           hovertemplate = paste("Winter Energy: %{y:$,.0f}<extra></extra>"))  
-    
+                           hovertemplate = paste("Winter Energy: %{y:$,.0f}<extra></extra>"))
+
     if("Best Start"  %in%  display_cols)
       p <- p %>% add_trace(data = Y, x = ~gross_wage1_annual, type = 'scatter', mode = 'none',
-                           y = ~BestStart_Total, 
-                           name = 'Best Start', fillcolor = set_colours[1], 
+                           y = ~BestStart_Total,
+                           name = 'Best Start', fillcolor = set_colours[1],
                            stackgroup = 'two',
-                           hovertemplate = paste("Best Start: %{y:$,.0f}<extra></extra>"))   
-    
-    
+                           hovertemplate = paste("Best Start: %{y:$,.0f}<extra></extra>"))
+
+
     #Adding a line for Net Income
     if("Net Income"  %in%  display_cols)
-      p <- 
-      p %>% add_lines(data = Y, x = ~gross_wage1_annual, 
+      p <-
+      p %>% add_lines(data = Y, x = ~gross_wage1_annual,
                       y = ~Net_Income, name = 'Net Income', color = I("black"),
-                      hovertemplate = paste("Net Income: %{y:$,.0f}<extra></extra>")) 
+                      hovertemplate = paste("Net Income: %{y:$,.0f}<extra></extra>"))
+    
+    # Add secondary axis with hours
+    p <- p %>% add_trace(x=~hours1, y=~0, line=list(width = 0), xaxis="x2",
+              data=X, showlegend=FALSE, inherit=FALSE,
+              hoverinfo = "none", type = "scatter", mode = "lines") %>%

```

Comment:
What does this pipe into?

Reply:
It pipes into the return(p) function; good catch